src/components/TrendingCard/__tests__/TrendingCard.spec.js

```diff
@@ -3,36 +3,63 @@
  * @copyright IBM Security 2020
  */
 
+import classnames from 'classnames';
 import { render } from '@testing-library/react';
 import React from 'react';
 
-import { Link } from '../../..';
-import { Default as TrendingCard } from '../TrendingCard.stories';
+import { Link, TrendingCard } from '../../../';
 
 describe('TrendingCard', () => {
-  it('renders', () => {
-    expect(render(<TrendingCard />).container.firstChild).toMatchSnapshot();
+  test('should have no Axe or DAP violations', async () => {
+    const main = document.createElement('main');
+    render(
+      <TrendingCard
+        // Use an empty `href` to avoid misdirected Axe "skip-link" violation:
+        href="#"
+        title="test title"
+        subtitle={<span>test subtitle</span>}
+      />,
+      {
+        // DAP requires a landmark '<main>' in the DOM:
+        container: document.body.appendChild(main),
+      }
+    );
+    await expect(document.body).toHaveNoAxeViolations();
+    await expect(document.body).toHaveNoDAPViolations('TrendingCard');
   });
 
-  it('renders a custom link', () => {
-    expect(
-      render(<TrendingCard element={Link} />).container.firstChild
-    ).toMatchSnapshot();
-  });
+  test('should accept a custom link', () => {
+    const { getByText, queryByTestId } = render(
+      <TrendingCard
+        title="test title"
+        className="test-trending-card-class"
+        element={({ children, className, ...other }) => (
```

Comment:
Would you mind clarifying on where these props are originating from? If I understand correctly, these will currently resolve as `undefined`

Reply:
It doesn't resolve to `undefined`... this enables you to use a custom component & pass through relevant props. 

It renders like this if you run `debug()` in this test:

```jsx
<body>
  <div>
    <a
      class="bx--link test-custom-link-class security--trending-card test-trending-card-class"
      data-testid="test-data-id"
      href="#test-href"
    >
      <h3
        class="security--trending-card__title"
      >
        test title
      </h3>
    </a>
  </div>
</body>
```

This passes through the `TrendingCard` children (i.e., "test title"), and it allows you to combine class names of `TrendingCard` AND the custom `Link`.

I can write a test for just `element={Link}` but this was meant to test how the custom element and the `TrendingCard` can share/combine certain things, like class name.

Reply:
Thank for clarifying!

Reply:
Hey @SimonFinney I believe an `elementType` can be an actual React component: https://reactjs.org/docs/typechecking-with-proptypes.html (also `https://github.com/facebook/prop-types/pull/211`)

And in this case, it is ideal to check like this because it is mimicking how one would pass in a custom link component from like a routing library, for example. They may not do just  `element={Link}`--  there may be a number of extra props they may need to pass through in their routing component, which is what this test is mimicking.

Do you think something is missing here?

Reply:
My understanding here is that `elementType` expects the element / component type, for example `Link` and not a component to be rendered, like `<Link prop="prop" />`.

This can be a standard HTML tag or a React component, as the internal logic of the `TrendingCard` generates the correct component and the props from that (See `createElement` - https://github.com/carbon-design-system/ibm-security/blob/dev/src/components/TrendingCard/index.js#L18).

This logic was replicated from how Carbon's `UIShell` handles its links - https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/UIShell/Link.js#L17

Let me know if you need further clarification!

Reply:
@SimonFinney I'm confused what your point is above? Let me put what I said in another way...

```jsx
// Defined in another file
const Link(() => <Link prop={prop} />);

// This test
<TrendingCard element={Link} />
```

is the same as 

```jsx
// This test
<TrendingCard element={({prop}) => <Link prop={prop} />
```

Here's another example in use, with `CustomLink`: https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/Button/Button-story.js#L118

Unless you are saying you want the test simplified? Please let me know.

Reply:
Exactly! Given the internal logic, we only need to ensure that the correct component is rendered